### ISSUE TITLE
gcp_resourcemanager_project error when deleting project

### ISSUE THREAD

**Issue by rgerardi** [ansible-collections/google.cloud#530]
##### SUMMARY
gcp_resourcemanager_project module throws `"Invalid JSON response with error: Expecting value: line 1 column 1 (char 0)"` error when deleting project

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
gcp_resourcemanager_project

##### ANSIBLE VERSION

```
ansible [core 2.14.1]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ricardo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/ricardo/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0] (/usr/bin/python)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```
# /usr/lib/python3.10/site-packages/ansible_collections
Collection   Version
------------ -------
google.cloud 1.0.2
```

Also tested with latest beta:
```
# /home/ricardo/.ansible/collections/ansible_collections
Collection   Version
------------ ------------
google.cloud 1.1.0-beta.0
```

##### CONFIGURATION

```
CONFIG_FILE() = /etc/ansible/ansible.cfg
```

##### OS / ENVIRONMENT
Arch Linux Kernel 6.0.11 all packages up-to-date

Also tested with RHEL 8.6 as well as running via ansible-navigator with custom execution environment

##### STEPS TO REPRODUCE
Execute playbook targeting existing GCP project. This example comes from the documentation.

```yaml
- name: Test project deletion
  hosts: localhost

  tasks:
    - name: Manage a project
      google.cloud.gcp_resourcemanager_project:
        id: "proj-1234"
        auth_kind: serviceaccount
        service_account_file: "sa-1234.json"
        parent:
          type: organization
          id: REDACTED
        state: "absent"
```

##### EXPECTED RESULTS
Project is deleted with no errors

##### ACTUAL RESULTS
Project is deleted but Ansible task throws a fatal error

```
The full traceback is:
  File "/tmp/ansible_google.cloud.gcp_resourcemanager_project_payload_b7pvn9xb/ansible_google.cloud.gcp_resourcemanager_project_payload.zip/ansible_collections/google/cloud/plugins
/modules/gcp_resourcemanager_project.py", line 326, in return_if_object
  File "/usr/lib/python3.10/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "auth_kind": "serviceaccount",
            "env_type": null,
            "id": "proj-1234",
            "labels": null,
            "name": null,
            "parent": {
                "id": "REDACTED",
                "type": "organization"
            },
            "project": null,
            "scopes": [
                "https://www.googleapis.com/auth/cloud-platform"
            ],
            "service_account_contents": null,
            "service_account_email": null,
            "service_account_file": "sa-1234.json",
            "state": "absent"
        }
    },
    "msg": "Invalid JSON response with error: Expecting value: line 1 column 1 (char 0)"
}
```

**Comment by toumorokoshi** [comment]
Thanks for the report! I think I have a repro and a fix. I'll try to have it in before the 1.1.0 release (ETA the next week or so)

**Comment by rgerardi** [comment]
Thank you. It's great to hear. Let me know if I can help testing it.

**Comment by Stephane-Br** [comment]
@toumorokoshi After a little debugging, I think the error should be around line [378](https://github.com/ansible-collections/google.cloud/blob/fe79a8b1e133c2b313f19818ace189cc63be5e86/plugins/modules/gcp_resourcemanager_project.py#L378) I think the condition is not ok

```
def wait_for_operation(module, response):
    op_result = return_if_object(module, response)
    if op_result is None:
        return {}
    status = navigate_hash(op_result, ['done'])
    wait_done = wait_for_completion(status, op_result, module)
    raise_if_errors(wait_done, ['error'], module)
    return navigate_hash(wait_done, ['response'])

```

`op_result` at that point == `{}`

```
(Pdb) op_result
{}
```
So it's not `None`

Could be changed to

`if op_result is None` -> `if not op_result`

**Comment by toumorokoshi** [comment]
@Stephane-Br thank you! I found the same thing, along with a follow-up fix for ensuring we detect the resource has already been soft-deleted.

I included that change in the commit as well, and tests now pass locally. You''ll see the fix in 1.1.0.

**Comment by Stephane-Br** [comment]
@toumorokoshi that was really fast, good job, we will wait for 1.1.0.

Thanks for your help
